dev/vcpkg/docker/Dockerfile

```diff
@@ -1,10 +1,6 @@
 FROM centos:7
 
-RUN sed -e 's|^mirrorlist=|#mirrorlist=|g' \
```

Comment:
Use an outdated repo.

There is repo fixup logic in `setup-build-depends.sh`, so we can remove this.

https://github.com/apache/incubator-gluten/blob/58907cfc5e3cf7d9d1fb15bd2ec0fbd4769548a0/dev/vcpkg/setup-build-depends.sh#L55-L59